src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Analyze/ConsoleAnalyze.cs

```diff
@@ -40,11 +44,12 @@ public async Task RunAsync(CancellationToken token)
             await _stateManager.LoadStateAsync(context, token);
             var analzyerContext = new AnalyzeContext(context);
             var analyzeResultMap = new List<AnalyzeResultDefinition>();
+
             foreach (var provider in _providers)
             {
                 analyzeResultMap.Add(new()
                 {
-                    Version = UpgradeVersion.Current.FullVersion,
+                    Version = GetProviderVersion(provider),
```

Comment:
I think the sarif log expects a value for the version, could you add a default value in case the GetProviderVersion returns empty?

Reply:
Good call. Defaulting to 0.0.0 in that case